azurerm/internal/services/dns/validate/dns_zone.go

```diff
@@ -0,0 +1,17 @@
+package validate
+
+import (
+	"fmt"
+	"regexp"
+)
+
+func DnsZoneSOARecordEmail(v interface{}, k string) (warnings []string, errors []error) {
+	value := v.(string)
+
+	if !regexp.MustCompile(`^[a-zA-Z\d_-][a-zA-Z\d._-]{0,62}$`).MatchString(value) {
+		errors = append(errors, fmt.Errorf("%q must be between 1 and 63 characters in length and contains only letters, numbers, underscores, dashes and periods", k))
+		return warnings, errors
+	}
+
+	return warnings, errors
+}
```

Comment:
Hi @neil-yechenwei - Is this correct?  My understanding it that this SOA value (RNAME) should be a valid email (with the `@` substituted with a `.`). This validation would prohibit the use of valid values since each dotted segment could be up to 63-characters in length. (to an email's max permitted length of 320 characters) Does the service put non-RFC restrictions on this value?

Reply:
These rules are from the error message from Azure portal. Updated rules.
![image](https://user-images.githubusercontent.com/19754191/99603632-ac2e8180-2a3e-11eb-8983-180b037c5544.png)
![image](https://user-images.githubusercontent.com/19754191/99603855-2b23ba00-2a3f-11eb-8efa-0c19780e1570.png)
![image](https://user-images.githubusercontent.com/19754191/99603914-4f7f9680-2a3f-11eb-9e2b-fcae037ec849.png)
![image](https://user-images.githubusercontent.com/19754191/99604852-67f0b080-2a41-11eb-8f32-f39100cd7e3d.png)
![image](https://user-images.githubusercontent.com/19754191/99611256-4cd86d80-2a4e-11eb-983f-5dbcfd9987e2.png)